### PR TITLE
9107 playlist name sanitization

### PR DESCRIPTION
**Pull request by tmtvl**
## Guidelines

1. Rebase before opening a pull request
2. If you are sending several unrelated fixes or features, use a branch and a separate pull request for each
3. If possible try squashing everything in a single commit. This is particularly beneficial in the case of feature merges since it allows easy bisecting when a problem arises

## Description

Implementing a filter for content labels in a playlist, as detailed in #9107 
This will allow for removal of parentheses (), brackets [], and the potential retention of regions (Australia, Europe, Japan, USA) and/or disc index.

## Related Issues

#9107

## Reviewers

@jdgleaver

### CODE REVIEW

**Comment by twinaphex** [comment]
Hi there,

looks like an interesting and promising PR!

@jdgleaver I'll let you review this one since you have taken such an active role as of late in pushing playlists forward.

**Comment by tmtvl** [comment]
Please be gentle, I have been doing a lot of Perl, so my C-fu isn't what it used to be.

**Comment by twinaphex** [comment]
Sure thing, we are more than appreciative for the work being rendered here and I am sure it will do just fine.

**Comment by orbea** [comment]
Please see the travis build logs, this breaks `C89_BUILD`, `CXX_BUILD`, xcode and android...

https://travis-ci.org/libretro/RetroArch/builds/564501021?utm_source=github_status&utm_medium=notification

File: libretro-common/playlists/label_sanitization.c [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+   bool copy = true;
+   int rindex = 0;
+   int lindex = 0;
+   char new_label[size];
+
+   for (; lindex < size && label[lindex] != '\0'; lindex++)
+   {
+      if (copy)
+      {
+         // check for the start of the range
```
Comment (orbea) [inline libretro-common/playlists/label_sanitization.c]: Please use C89 compatible comments.

**Comment by orbea** [comment]
For CXX_BUILD.
```
menu/cbs/menu_cbs_left.c: In function ‘int playlist_label_display_mode_left(unsigned int, const char*, bool)’:
menu/cbs/menu_cbs_left.c:448:25: warning: no ‘operator--(int)’ declared for postfix ‘--’, trying prefix operator instead [-fpermissive]
       label_display_mode--;
                         ^
menu/cbs/menu_cbs_left.c:448:25: error: no match for ‘operator--’ (operand type is ‘playlist_label_display_mode’)
Makefile:202: recipe for target 'obj-unix/release/menu/cbs/menu_cbs_left.o' failed
make: *** [obj-unix/release/menu/cbs/menu_cbs_left.o] Error 1
```

**Comment by orbea** [comment]
For android and xcode you want to update `griffin/griffin.c`.

File: libretro-common/playlists/label_sanitization.c [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+bool left_parens_or_brackets(char *left)
+{
+   return left[0] == '(' || left[0] == '[';
+}
+
+bool right_parens_or_brackets(char *right)
+{
+   return right[0] == ']' || right[0] == ']';
```
Comment (RobLoach) [inline libretro-common/playlists/label_sanitization.c]:
``` suggestion
   return right[0] == ')' || right[0] == ']';
```

File: libretro-common/playlists/label_sanitization.c (hunk 1 above)
Comment (RobLoach) [inline libretro-common/playlists/label_sanitization.c]:
``` suggestion
         /* Check for the start of the range */
```

File: libretro-common/playlists/label_sanitization.c [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+      else
+         return true;
+   }
+   else
+      return false;
+}
+
+void label_default_display(char *label, size_t size)
+{
+   return;
```
Comment (RobLoach) [inline libretro-common/playlists/label_sanitization.c]: Would be good to have a comment about not doing anything here.

``` suggestion
   /* The default display will keep the label the same. */
   return;
```

File: libretro-common/playlists/label_sanitization.c [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+            && left[5] == 'p'
+            && left[6] == 'e')
+         || (left[1] == 'J'
+            && left[2] == 'a'
+            && left[3] == 'p'
+            && left[4] == 'a'
+            && left[5] == 'n')
+         || (left[1] == 'U'
+            && left[2] == 'S'
+            && left[3] == 'A'))
```
Comment (RobLoach) [inline libretro-common/playlists/label_sanitization.c]: Is there a libretro-common string function we could use for this? Hmm, perhaps not? Seems like a substring compare would be handy to use elsewhere too.

**Comment by RobLoach** [comment]
Nice work! Coming along well. Was able to hide the tags :smiley: 

When I removed both params and brackets in Playstation, I found the .lpl file got the following written in the JSON:
```
  "label_display_mode": 3,
```

Thumbnails still work with this too :+1: 

Screenshot at 2019-07-27 21-14-51

Looking great! Once we clean up a few things, and goes through a few reviews, this will be a great addition for 1.7.8.

File: libretro-common/playlists/label_sanitization.c (hunk 4 above)
Comment (tmtvl) [reply libretro-common/playlists/label_sanitization.c]: Yeah, strncmp would work. I should probably also write a test for the sanitization functions.

**Comment by twinaphex** [comment]
BTW,

try to use our strcmp/strncmp wrappers in libretro-common/include/string/stdstring.h instead. 

string_is_equal and friends.

**Comment by tmtvl** [comment]
I'll close the PR while I work on the fixes.

**Pull request closed without merging**